src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -182,6 +187,9 @@ public bool SetNullableContext(byte? value)
             TupleExtraData? tupleData = null)
             : base(tupleData)
         {
+            // If we're dealing with a simple program, then we must be in the global namespace
+            Debug.Assert(containingSymbol is NamespaceSymbol { IsGlobalNamespace: true } || !declaration.Declarations.Any(d => d.IsSimpleProgram));
```

Comment:
Is it true that we can't get here in an error scenario like the following:

```cs
namespace NS;

var x = 1;
```

Reply:
Correct. I'll add a few tests.
`ParseNamespaceBody` parses differently in global context (outside of a namespace). See `isGlobal` local. It recognizes when inside a namespace (either with brace or semi-colon).